server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java

```diff
@@ -243,12 +240,12 @@ public long getRefreshSeqNoLag() {
         return refreshSeqNoLag;
     }
 
-    private void computeTimeMsLag() {
-        timeMsLag = localRefreshTimeMs - remoteRefreshTimeMs;
-    }
-
     public long getTimeMsLag() {
-        return timeMsLag;
+        if (remoteRefreshTimeMs == localRefreshTimeMs) {
+            logger.info("remoteRefreshTimeMs={} localRefreshTimeMs={}", remoteRefreshTimeMs, localRefreshTimeMs);
+            return 0;
+        }
+        return currentTimeMsUsingSystemNanos() - localRefreshTimeMs;
```

Comment:
What if remote refresh hasn't happened yet but local refresh has, wouldn't this unnaturally spike during that interval

Reply:
@Bukhtawar no, we are going to track now the time elapsed from the time the document visible on the primary shard.